src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -22,18 +30,23 @@ export const TodoFilter: React.FC = () => {
           type="text"
           className="input"
           placeholder="Search..."
+          value={query}
+          onChange={(e) => dispatch(actions.setQuery(e.target.value))}
         />
         <span className="icon is-left">
           <i className="fas fa-magnifying-glass" />
         </span>
 
         <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button
-            data-cy="clearSearchButton"
-            type="button"
-            className="delete"
-          />
+          {!!query && (
+            /* eslint-disable-next-line jsx-a11y/control-has-associated-label */
```

Comment:
You have a linting rule to always have a label associated with form control. Though you've disabled it for the button, it's better to have a visually hidden label for accessibility.